packages/libp2p/src/circuit-relay/utils.ts

```diff
@@ -13,6 +12,8 @@ import type { Uint8ArrayList } from 'uint8arraylist'
 const log = logger('libp2p:circuit-relay:utils')
 
 async function * countStreamBytes (source: Source<Uint8Array | Uint8ArrayList>, limit: { remaining: bigint }): AsyncGenerator<Uint8Array | Uint8ArrayList, void, unknown> {
```

Comment:
we probably should change the signature of this function to

```suggestion
async function * countStreamBytes (source: Source<Uint8Array | Uint8ArrayList>, limitBytes: bigint): AsyncGenerator<Uint8Array | Uint8ArrayList, void, unknown> {
```

and replace all the `limit.remaining`

Reply:
Unfortunately this doesn't work and would introduce a new bug.

The data limit applies in both directions (e.g. `A->Relay->B` and `B->Relay->A`) so both ends of the stream need to decrement the same `remaining` variable.

Passing an object copies by reference, passing a number would copy by value so you'd end up with separate limits for each direction.